### ISSUE TITLE
Few Issues observed in this project (Effort to make it perfect)

### ISSUE THREAD

**Issue by 00itsm00** [D3VL/L3MON#84]
**Hello There,**
I would like to share my exprience but firstly i would like to THANK dev team who made the effort to spare their precious time to actually making this project possible.

**Now Dear All,**

I tried and launched L3MON on 4 of servers and almost 8 to 10 times & they worked perfectly fine without any issues.
![1](https://user-images.githubusercontent.com/26104191/76893040-1d60d800-68a5-11ea-84f0-9b5114e737bc.jpg)
Few things i would like to share here which needs to be fixed / updated:

1. In devices there's no button to remove / block any device.
![clipb](https://user-images.githubusercontent.com/26104191/76893447-cad3eb80-68a5-11ea-8e79-98709c249af6.JPG)

2. No update button in clipboard & notification.

3. Please update selection / select all check boxes in each side menu so that we ll be able to select desired or all the things to download.

4. In event Log there's no button to clear the logs or export any log. (desired ones with check boxes or all) Export should be in Excel or PDF or whatever you prefer.

5. Multi User is a plus if you can add this option with separate profile e.g all devices should not be shown in every user (hope you get it) but admin user.

**I hope this time i posted in correct section.**

I ll report more issues which i ll face while playing with it alongwith suggestions.

Cheers!!.

**Comment by Jack-Rogers** [comment]
Thank you so much for your positive words! It's easy to make a comment when you have an issue, but for the 99% of people where it works flawlessly, they never think to leave a comment! It's great to hear! 

To address your requests;
1) Please see issue #36 & #45 for removing devices, it's a planned feature 👌 
2) They don't need update buttons! they're initiated by the device, instead of polling the device for this information, the device sends it to the server the second it is available 😄 
3) We like this idea of bulk downloading, something we can look into implementing ⬇️ 
4) Sorting of logs to specific types could be useful, a previous request has been made for exporting information, something else to be looked at 🔎 
5) Multi-user is potentially coming, though we're not sure how we're going to release that yet! Hang tight, ill pop a message here when we have decided 😎 

Any more issues or requests please open up a new issue 😄 
And yep, 100% the right place now, Thanks!

**Comment by 00itsm00** [comment]
> Thank you so much for your positive words! It's easy to make a comment when you have an issue, but for the 99% of people where it works flawlessly, they never think to leave a comment! It's great to hear!
> 
> To address your requests;
> 
> 1. Please see issue #36 & #45 for removing devices, it's a planned feature 👌
> 2. They don't need update buttons! they're initiated by the device, instead of polling the device for this information, the device sends it to the server the second it is available 😄
> 3. We like this idea of bulk downloading, something we can look into implementing ⬇️
> 4. Sorting of logs to specific types could be useful, a previous request has been made for exporting information, something else to be looked at 🔎
> 5. Multi-user is potentially coming, though we're not sure how we're going to release that yet! Hang tight, ill pop a message here when we have decided 😎
> 
> Any more issues or requests please open up a new issue 😄
> And yep, 100% the right place now, Thanks!

Hi Jack,

Need a suggestion.

My account got blocked by provider due to suspicious traffic and they send me the logs also which i am attaching here.
![image](https://user-images.githubusercontent.com/26104191/77087599-be25d380-6a1c-11ea-9e26-f3e1fc47e298.png)
is there anything which can be done to hide or bypass this so that the ip or account ll not be blocked!!.

Thanks.

**Comment by Jack-Rogers** [comment]
Hmm, that doesn't look like the sort of traffic L3MON will generate, was there anything else running on your VPS?
As a side note, we highly recommend using OVH for VPS, We've never had an issue with them and regularly use them for all our hosting needs 👍

**Comment by 00itsm00** [comment]
Hi Jack,

Thanks for your reply, i bought 4 servers specifically to run / test it. till now only one on them went down, and thanks for the suggestion i ll definitely give a try to OVH.

Please also share how to configure and install Nginx server as it has been recommended in the readme.

Thank you for the support.

Regards,

itsM

**Comment by 00itsm00** [comment]
Hi @Jack-Rogers hope you are doing well,

i am having issue on windows server 2012 while building apk, once i click build apk it shows that it is building but at finalizing it show below error, maybe the permission error.

![Error at finalizing](https://user-images.githubusercontent.com/26104191/78032766-90268480-7376-11ea-8eee-5e085e995c8a.jpg)

Could you please guide me on that.

Thank you.

**Comment by 00itsm00** [comment]
Hi Jack,

Hope you are good.

Would like to share something which i observed just right now.

I freshly installed it on a server and build apk it was all success, i only buld the apk not installed or even copied in any device but it shows in logs install successful alongwith a device which popped up in device manager.

I think whoever installs apk in any device it shows in all users interface whoever have installed it (which is live).

Could you please explain what is the issue and why we are seeing this and please attenpt to solve it.

Images are being attached for reference:

![tempsnip2](https://user-images.githubusercontent.com/26104191/81470573-78a3ac80-91fc-11ea-9909-42f54e0384b5.jpg)
![tempsnip](https://user-images.githubusercontent.com/26104191/81470577-79d4d980-91fc-11ea-883e-06de8ea61387.jpg)

**Comment by 00itsm00** [comment]
Hi Team,

Reinstalled on another server and after leaving it untouched for couple of days now again getting unknown devices in device manager.

Please look into it.
Not able to post screencaps because i am posting from smartphone.
Thanks

**Comment by 00itsm00** [comment]
Dear Team,

Greetings,
Posting above issue again.

installed it on server and left it for almost 2 or 3 weeks.

devices were being added automatically, was getting logs too.

Below images are for reference:

![image](https://user-images.githubusercontent.com/26104191/84412311-1b967e80-ac20-11ea-934e-622ef1381b7b.png)


![image](https://user-images.githubusercontent.com/26104191/84412221-face2900-ac1f-11ea-8316-e6d7cc687fe2.png)

Thanks.

**Comment by Jack-Rogers** [comment]
Thanks @00itsm00 
Where is this server from - all I can think of is someone else has used the same IP to run L3MON on before and thus those devices are connecting to your instance.

The only possible way to see random devices connecting to your server is if there's an APK out there with your servers IP in it. 

Was this behaviour observed with 2 different servers with different IP's?

**Comment by 00itsm00** [comment]
Hi Jack,

Thank you for your reply.

I have observed it since long and tried it on different servers.

I strongly feel that its a bug.

Whenever i install it on any of my servers after sometime it shows devices listed in it.

I uninstalled it already i ll do a fresh install again after sometime and if you need logs or access to the server drop me an email @ itsm@gmx.se or ping me at whatsapp +17868861212 i ll be glad to hear from you.

Thanks.

Regards,

itsM

**Comment by Jack-Rogers** [comment]
Let's keep the conversation here for the minute 👍

Please can you install it on a new server, preferably one that is not linked to any current infrastructure of the server you've been having this issue with so we can eliminate it being some odd setup issue.

Additionally, are you using a hostname or direct IP access? 

I can't see how this could possibly be a bug as users are connecting to your server, maybe you have some suggestions or ideas as to how this could be happening as the app is made to only point to your own server, not anybody else's! I suspect that maybe you've released a version of this before and either have a hostname pointed to the servers IP or you're simply just using the same IP as a previous build.

Cheers,
Jack

**Comment by 00itsm00** [comment]
Hi Jack,

I ll install it on a fresh purchase ll try to do it today while i ll be in office.

Moreover, i have tried it multiple times before reporting & i am using with direct ip access.

Suggestions:
What i think (think not sure) that it is adding devices it self whenever any of user is adding a device it shares some of the information.

It is not giving any logs like pictures apps recording etc, but it shares only connect disconnect time alongwith location and timestamps of connect and disconnect when user connects and time of connection on is more likely 1 minute to 4 or 5 minutes not more than that also it queues the command for update but not updating anything.

I think it's enough info for now.

ll share more after observing its behavior for a while.

Thank you for giving some time here.

Cheers,
itsM
